src/roles/index.ts

```diff
@@ -10,37 +10,29 @@ import {
   PermissionsInput,
   Role,
   RolesPermission,
-  User,
 } from './types.js';
 import { Map } from './util.js';
 
 export interface Roles {
   listAll: () => Promise<Record<string, Role>>;
-  ofCurrentUser: () => Promise<Record<string, Role>>;
   byName: (roleName: string) => Promise<Role | null>;
-  byUser: (user: string) => Promise<Record<string, Role>>;
-  assignedUsers: (roleName: string) => Promise<Record<string, User>>;
+  assignedUserIds: (roleName: string) => Promise<string[]>;
   delete: (roleName: string) => Promise<void>;
   create: (roleName: string, permissions: PermissionsInput) => Promise<Role>;
-  assignToUser: (roleNames: string | string[], user: string) => Promise<void>;
   exists: (roleName: string) => Promise<boolean>;
-  revokeFromUser: (roleNames: string | string[], user: string) => Promise<void>;
   addPermissions: (roleName: string, permissions: PermissionsInput) => Promise<void>;
   removePermissions: (roleName: string, permissions: PermissionsInput) => Promise<void>;
-  hasPermission: (roleName: string, permission: Permission) => Promise<boolean>;
+  hasPermissions: (roleName: string, permission: Permission) => Promise<boolean>;
```

Comment:
```suggestion
  hasPermissions: (roleName: string, permission: Permission | Permission[]) => Promise<boolean>;
```

suggestion(blocking): I know TS has no problem with the implementation accepting a wider type than the interface declares, but this is probably truer to the method's name